### PR TITLE
Added TestSourceProcessor helper

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Added TestSourceProcessor which is similar to TestProcessor but is specialized for testing Sources.

#### PR validation:

Code compiles and new unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
Unit test failure looks related

**Comment by Dr15Jones** [comment]
please test

File: FWCore/TestProcessor/src/oneTimeInitialization.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#if !defined(FWCore_TestProcessor_oneTimeInitialization_h)
```
Comment (makortel) [inline FWCore/TestProcessor/src/oneTimeInitialization.h:1]: Could this be 
```suggestion
#ifndef FWCore_TestProcessor_oneTimeInitialization_h
```
?

**Comment by Dr15Jones** [comment]
hold

additional testing using PoolSource has shown problems.

**Comment by Dr15Jones** [comment]
unhold

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
The unit test failure during the test appears to be from a failed cmsDriver.py command. That could not have been caused by this PR.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/39803 and 29634.911 (known non-reproducibility, but without a GH issue)

**Comment by makortel** [comment]
+core

**Comment by srimanob** [comment]
Hi @makortel 
Strange that the fix is on HGCal, but the failures are not on HGCal at all. Any possibility that somethings come between CMSSW_14_1_X_2024-07-15-2300 and now ?

**Comment by makortel** [comment]
> Any possibility that something comes between CMSSW_14_1_X_2024-07-15-2300 and now ?

Of course that is possible, but I'd suspect more an existing issue that was hidden by the HGCal uninitialized variables. For the record, here are the PRs merged after 07-15-2300 up to 07-18-1100
* https://github.com/cms-sw/cmssw/pull/45470
* https://github.com/cms-sw/cmssw/pull/45446
* https://github.com/cms-sw/cmssw/pull/45366
* https://github.com/cms-sw/cmssw/pull/45428
* https://github.com/cms-sw/cmssw/pull/45465
* https://github.com/cms-sw/cmssw/pull/45307
* https://github.com/cms-sw/cmssw/pull/45375
* https://github.com/cms-sw/cmssw/pull/45452
* https://github.com/cms-sw/cmssw/pull/45409
* https://github.com/cms-sw/cmssw/pull/45424
* https://github.com/cms-sw/cmssw/pull/45437
* https://github.com/cms-sw/cmssw/pull/45444
* https://github.com/cms-sw/cmssw/pull/45445
* https://github.com/cms-sw/cmssw/pull/45447
* https://github.com/cms-sw/cmssw/pull/45455
* https://github.com/cms-sw/cmssw/pull/45457
* https://github.com/cms-sw/cmssw/pull/45471

**Comment by rappoccio** [comment]
+1

**Pull request merged**